bondjimbond/islandora_badges
# Super fun happy everything.

This PR subsumes #18 

Resolves #17, resolves #19 

# What does this Pull Request do?

Splits the Altmetrics, Scopus and Web of Science functionality into separate sub-modules of the islandora_badges module.

Allows the configuration of which content-models the badges should appear (currently only implemented on islandora_altmetrics block).

Allows viewing a sample altmetrics badge in configuration to see what it will look like.

# What's new?

* Separate modules for each badge, with their own configuration.
* Shared DOI extraction function and (possibly) the content-model filtering.
* Removed islandora_scholar dependency.
* Removed defaults values and switched to empty strings.

# How should this be tested?

1. Enable the Islandora Badges, Islandora Altmetrics, Islandora Scopus & Islandora WOS modules.
1. Add the 3 blocks to your display.
1. Goto Admin > Islandora > Islandora Utility Modules > Islandora Badges Configuration
1. Select one or more content-model to display badges on. 
1. Save config.
1. Add your Scopus API Key on its config page. Save config.
1. Add your Web Of Science username/password on its config page. Save config.
1. Configure your Altmetrics badge on its config page. Save config.
1. Browse to your objects and see that the badge shows up on the correct content-model and displaying correctly.

**TODO**: This needs an update hook to convert the islandora_badges_altmetrics_... variables back to islandora_altmetrics_... variables.

Example:
* Does this change require documentation to be updated? yes
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? possibly.

# Interested parties
@bondjimbond @mjordan
